packages/flutter/test/widgets/custom_paint_test.dart

```diff
@@ -77,4 +78,27 @@ void main() {
     expect(target.currentContext.size, Size.zero);
 
   });
+
+  testWidgets('Raster cache hints', (WidgetTester tester) async {
+    final GlobalKey target = new GlobalKey();
+
+    final List<String> log = <String>[];
+    await tester.pumpWidget(new CustomPaint(
+      key: target,
+      isComplex: true,
+      painter: new TestCustomPainter(log: log),
+    ));
+    RenderCustomPaint renderCustom = target.currentContext.findRenderObject();
+    expect(renderCustom.isComplex, true);
+    expect(renderCustom.willChange, false);
+
+    await tester.pumpWidget(new CustomPaint(
+      key: target,
+      willChange: true,
+      foregroundPainter: new TestCustomPainter(log: log),
+    ));
+    renderCustom = target.currentContext.findRenderObject();
+    expect(renderCustom.isComplex, false);
+    expect(renderCustom.willChange, true);
+  });
```

Comment:
bonus points would be awarded for a test that checks that the bits actually get set properly all the way to the layer tree :-)